SConstruct

```diff
@@ -38,63 +38,67 @@ AddOption(
     action='store_true',
     help='msvc12',
     default=False)
-#run scons --sdk in order to build the sdk
-AddOption(
-    '--sdk',
-    action='store_true',
-    help='sdk',
-    default=False)
 #run scons --samples in order to build the samples
 AddOption(
     '--samples',
     action='store_true',
     help='samples',
     default=False)
-
-isX86 = GetOption('x86')
-build_arch = "x86" if isX86 else "x64"
-isRelease = GetOption('release')
-build_flavor = "release" if isRelease else "debug"
+#run scons --qt in order to specify the qt installation path
+AddOption(
+    '--qt',
+    type='string',
+    nargs=1,
+    action='store',
+    help='qt',
+    metavar='DIR')
+
+build_arch = GetOption('arch')
+build_flavor = GetOption('configuration').upper()
 platform = sys.platform
 msvc12 = GetOption('msvc12')
-sdk = GetOption('sdk')
 samples = GetOption('samples')
+qt_dir = GetOption('qt')
 
-# TODO: move to build_config_<platform>
-if isX86:
-  if platform == 'win32':
-    target_arch = "x86"
-    arch_suffix = ''
-    win_def = 'WIN32'
-  else:
-    target_arch = 'i386'
+if platform == 'win32':
+    from build_config_win32 import *
+elif platform == 'linux2':
+    from build_config_linux import *
+
+if not qt_dir:
+    qt_dir = QT_DIR_DEFAULT 
+
+ccflags = CCFLAGS + ' ' + eval("CCFLAGS_" + build_arch) + ' ' + eval("CCFLAGS_" + build_flavor)
+cxxflags = CXXFLAGS + ' ' + eval("CXXFLAGS_" + build_arch) + ' ' + eval("CXXFLAGS_" + build_flavor)
```

Comment:
compiling with x86 causes the error:
NameError: name 'CCFLAGS_x86' is not defined

Reply:
you have defined CCFLAGS_x32...

---
In reply to: [140315640](https://github.com/AzureAD/rms-sdk-for-cpp/pull/162#discussion_r140315640) [](ancestors = 140315640)